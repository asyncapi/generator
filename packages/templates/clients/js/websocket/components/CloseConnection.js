import { Text } from '@asyncapi/generator-react-sdk';

export function CloseConnection() {
  return (
    <Text indent={2}>
      {
        `// Method to close the WebSocket connection
close() {
  if (this.websocket) {
    this.websocket.close();
    console.log('WebSocket connection closed.');
  }
}`
      }
    </Text>
  );
}
