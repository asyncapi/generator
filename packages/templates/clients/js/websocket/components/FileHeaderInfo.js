import { Text } from '@asyncapi/generator-react-sdk';

export function FileHeaderInfo({ info, server }) {
  return (
    <Text>
      <Text>
        {'//////////////////////////////////////////////////////////////////////'}
      </Text>
      
      <Text>
        {'//'}
      </Text>

      <Text>
        {'//'} {info.title()} - {info.version()}
      </Text>

      <Text>
        {'//'} Protocol: {server.protocol()}
      </Text>

      <Text>
        {'//'} Host: {server.host()}
      </Text>

      {server.hasPathname() && (
        <Text>
          {'//'} Path: {server.pathname()}
        </Text>
      )}

      <Text>
        {'//'}
      </Text>

      <Text>
        {'//////////////////////////////////////////////////////////////////////'}
      </Text>
    </Text>
  );
}