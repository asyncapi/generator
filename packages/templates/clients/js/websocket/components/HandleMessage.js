import { Text } from '@asyncapi/generator-react-sdk';

export function HandleMessage() {
  return (
    <Text newLines={2} indent={2}>
      {
        `// Method to handle message with callback
handleMessage(message, cb) {
  if (cb) cb(message);
}`
      }
    </Text>
  );
}
