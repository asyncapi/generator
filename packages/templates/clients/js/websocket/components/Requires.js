import { Text } from '@asyncapi/generator-react-sdk';

export function Requires() {
  return (
    <Text>
      <Text>
        const WebSocket = require('ws');
      </Text>
    </Text>
  );
}

