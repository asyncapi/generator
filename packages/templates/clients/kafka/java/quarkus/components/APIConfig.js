import { Text } from "@asyncapi/generator-react-sdk";


export default function APIConfig({ protocol }) {
    if (!protocol) {
        throw new Error('Protocol is not defined.');
    }

    const protocolEnum = protocol.toLowerCase();
    const templateType = `${protocolEnum}-template`;
    const configProperties = {
        "wss": `compression:
                    type: boolean
                    default: true
                  subprotocols:
                    type: array
                    items:
                      type: string
                  maxConnections:
                    type: integer
                    default: 100`,
        "kafka": ` # Kafka specific
                  partitions:
                    type: integer
                    default: 3
                  replicationFactor:
                    type: integer
                    default: 1
                  retentionMs:
                    type: integer
                    default: 604800000 # 7 days`
    };
    const topicORPathProperties = {
        "wss": `path:
                  type: string
                  description: "Path to the API endpoint."`,
        "kafka": `topic:
                  type: string
                  description: "Kafka/MQTT topic name for the API."`

    } 

    return (
        <Text>
            {`
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  name: apiconfigs.asyncapi.com
spec:
  group: asyncapi.com
  scope: Namespaced
  names:
    plural: apiconfigs
    singular: apiconfig
    kind: APIConfig
  versions:
    - name: v1
      served: true
      storage: true
      schema:
        openAPIV3Schema:
          type: object
          properties:
            spec:
              type: object
              required:
                - protocol
                - templateType
              properties:
                protocol:
                  type: string
                  enum: ["${protocolEnum}"]
                  description: "Message protocol used by the API, e.g., kafka, mqtt, etc."
                ${topicORPathProperties[protocolEnum]}
                templateType:
                  type: string
                  enum: ["${templateType}"]
                config:
                  type: object
                  properties:
                    replicas:
                      type: integer
                      minimum: 1
                      default: 1
            status:
              type: object
              properties:
                deployed:
                  type: boolean
                phase:
                  type: string
                  enum: ["Pending", "Creating", "Running", "Failed", "Deleting"]
                  description: "Current lifecycle phase of the APIConfig."
                message:
                  type: string
                url:
                  type: string
                  description: "Service endpoint URL"
                lastReconciled:
                  type: string
                  format: date-time`}
        </Text>
    );

}

                                              

/**
 * path:
                  type: string
                  description: "Path to the API endpoint." # this is for websockets
                topic:
                  type: string
                  description: "Kafka/MQTT topic name for the API."
apiVersion: apiextensions.k8s.io/v1
            kind: CustomResourceDefinition
            metadata:
                name: apiconfigs..asyncapi.com
            spec:
                group: .asyncapi.com
                versions:
                    - name: v1
                      served: true
                      storage: true
                      schema:
                          openAPIV3Schema:
                              type: object
                              properties:
                                  spec:
                                      type: object
                                      required:
                                      -protocol
                                      -templateType
                                      properties:
                                          protocol:
                                              type: string
                                              enum: ["{protocol}"]
                                              description: "Message protocol used by the API, e.g., kafka, mqtt, etc."
                                          path:
                                              type: string
                                              description: "Path to the API endpoint."
                                          topic:
                                              type: string
                                              description: "Kafka/MQTT topic name for the API."
                                          templateType:
                                              type: string
                                              enum: ["{templateType}"]
                                          config:
                                              type: object
                                              properties:
                                                {configProperties[protocolEnum]}
                                                replicas:
                                                    type: integer
                                                    default: 1
                                                resources:
                                                    type: object
                                                    properties:
                                                    requests:
                                                        type: object
                                                        properties:
                                                        memory:
                                                            type: string
                                                            default: "128Mi"
                                                        cpu:
                                                            type: string
                                                            default: "100m"
                                                    limits:
                                                        type: object
                                                        properties:
                                                        memory:
                                                            type: string
                                                            default: "512Mi"
                                                        cpu:
                                                            type: string
                                                            default: "500m"
                              status:
                                type: object
                                properties:
                                deployed:
                                    type: boolean
                                phase:
                                    type: string
                                    enum: ["Pending", "Creating", "Running", "Failed", "Deleting"]
                                message:
                                    type: string
                                url:
                                    type: string
                                    description: "Service endpoint URL"
                                lastReconciled:
                                    type: string
                                    format: date-time
                                            
                scope: Namespaced
                names:
                    plural: apiconfigs
                    singular: apiconfig
                    kind: APIConfig
                    shortNames:
                    - apicfg
 */