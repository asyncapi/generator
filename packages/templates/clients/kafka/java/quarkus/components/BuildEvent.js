import { Text } from "@asyncapi/generator-react-sdk";
import { FormatHelpers } from "@asyncapi/modelina";


function generateAddMethodCalls(headerInfo) {
  return headerInfo.map((header, index) => {
    // Convert header name to camelCase (or whatever format you prefer for variable names)
    const variableName = FormatHelpers.toCamelCase(header.schemaId);
    const indentLevel = index === 0 ? '' : '\t\t\t\t\t\t\t\t\t\t\t  ';

    
    // Return the formatted .add() method call
    return `${indentLevel}.add("${header.name}", ${variableName}.getBytes())`;
  }).join('\n'); // Join the generated .add() calls with new lines
}


export default function BuildEvent({ headerInfo }){

    const addMethodCalls = generateAddMethodCalls(headerInfo);
    console.log("Add Method Calls: ", addMethodCalls);

    return (
        <Text indent={2} newLines={2}>
            {`
    try{
            String payloadJson = new ObjectMapper().writeValueAsString(payload);

            // Create metadata with headers
            OutgoingKafkaRecordMetadata<String> metadata = OutgoingKafkaRecordMetadata.<String>builder()
                .withHeaders(new RecordHeaders()
                        ${addMethodCalls})
                .build();

            org.eclipse.microprofile.reactive.messaging.Message<String> message = KafkaRecord.of(requestId, payloadJson)
                .addMetadata(metadata); // not sure about this payload thing !!! ask lukasz

                `}
        </Text>
    );


}