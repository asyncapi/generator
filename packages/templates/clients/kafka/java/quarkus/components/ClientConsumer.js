import ConsumeEvent from './ConsumeEvent';
import { ConsumerFields } from './ConsumerFields';
import { Text } from '@asyncapi/generator-react-sdk';

export default function ClientConsumer({ className, eventName }) {
  const formattedEventName = eventName ? eventName.replace(/^\w/, (c) => c.toUpperCase()) : eventName;

  return (
    <Text newLines={2}>
      <Text newLines={2}>
        {`
@ApplicationScoped  
public class ${className}{`}
      </Text>
      <ConsumerFields clientName={className}/>
      <ConsumeEvent eventName={formattedEventName}/>
      
    </Text>
  );
}
