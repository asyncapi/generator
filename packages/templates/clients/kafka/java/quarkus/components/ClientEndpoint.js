import { Text } from '@asyncapi/generator-react-sdk';
import { EndpointFields } from './EndpointFields';
import { ProduceEvent } from './ProduceEvent';
import { MiddleService } from './MiddleService';

export default function ClientEndpoint({ className, operations, channels, correctPath }) {
  if (!correctPath) {
    correctPath = '/';
  }
    
  const sendOperations = operations.filterBySend();

  return (
    <Text newLines={2}>
      <Text newLines={2}>
        {`
@Path("${correctPath}")  
@Consumes(MediaType.APPLICATION_JSON)
@Produces(MediaType.APPLICATION_JSON)
public class ${className}{`}
      </Text>
      <EndpointFields sendOperations={sendOperations} channels={channels} />
      <ProduceEvent sendOperations={sendOperations}/>
      <MiddleService channels={channels} />
      <Text newLines={2}>
        {`
}`}
      </Text>
        
    </Text>
  );
}
