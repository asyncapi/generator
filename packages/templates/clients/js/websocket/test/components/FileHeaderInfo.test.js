import { FileHeaderInfo } from '../../components/FileHeaderInfo';

describe('testing of FileHeaderInfo function', () => {
  jest.setTimeout(100000);

  const mockData = {
    info: {
      title: () => 'Test AsyncAPI Service',
      version: () => '1.0.0'
    },
    server: {
      protocol: () => 'mqtt',
      host: () => 'test.mosquitto.org',
      pathname: () => '/mqtt',
      hasPathname: () => true
    },
    serverWithoutPath: {
      protocol: () => 'mqtt',
      host: () => 'test.mosquitto.org',
      pathname: () => '',
      hasPathname: () => false
    }
  };

  test('render websockets file header info correctly', () => {
    const wrapper = FileHeaderInfo({ info: mockData.info, server: mockData.server });
    expect(wrapper).toMatchSnapshot();
  });

  test('render websockets file header when path is missing', () => {
    const wrapper = FileHeaderInfo({ info: mockData.info, server: mockData.serverWithoutPath });
    expect(wrapper).toMatchSnapshot();
  });
});
