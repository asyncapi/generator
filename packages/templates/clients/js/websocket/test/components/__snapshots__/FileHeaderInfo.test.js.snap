// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of FileHeaderInfo function render websockets file header info correctly 1`] = `
<Text>
  <Text>
    //////////////////////////////////////////////////////////////////////
  </Text>
  <Text>
    //
  </Text>
  <Text>
    //
     
    Gemini Market Data Websocket API
     - 
    1.0.0
  </Text>
  <Text>
    //
     Protocol: 
    wss
  </Text>
  <Text>
    //
     Host: 
    api.gemini.com
  </Text>
  <Text>
    //
     Path: 
    /v1/marketdata
  </Text>
  <Text>
    //
  </Text>
  <Text>
    //////////////////////////////////////////////////////////////////////
  </Text>
</Text>
`;

exports[`Testing of FileHeaderInfo function render websockets file header info correctly 2`] = `
<Text>
  <Text>
    //////////////////////////////////////////////////////////////////////
  </Text>
  <Text>
    //
  </Text>
  <Text>
    //
     
    Gemini Market Data Websocket API
     - 
    1.0.0
  </Text>
  <Text>
    //
     Protocol: 
    wss
  </Text>
  <Text>
    //
     Host: 
    api.gemini.com
  </Text>
  <Text>
    //
  </Text>
  <Text>
    //////////////////////////////////////////////////////////////////////
  </Text>
</Text>
`;
